aws-amplify/docs
# Handle links with specified platform and add tests

#### Description of changes:
- Added logic to not add the `query` object if the platform was already specified in url when trying to set up an MDX link
  - Previously if we linked directly to a platform, it would add the platform as a query param (`?platform=react`)

To test:
1. Go to https://fix-specific-platform-linking.d1ywzrxfkb9wgg.amplifyapp.com/flutter/build-a-backend/existing-resources/cli/ 
2. In the callout, `Connecting a web app?...`, click on "React guide"
3. Verify that there isn't a platform query added to the url
4. Click around the site to test links
   1. Check hash link urls
      1.  https://fix-specific-platform-linking.d1ywzrxfkb9wgg.amplifyapp.com/gen2/reference/telemetry/ - The "opt out" link at the top of the page
      2.  https://fix-specific-platform-linking.d1ywzrxfkb9wgg.amplifyapp.com/react/build-a-backend/troubleshooting/upgrade-amplify-packages/#check-for-duplicate-versions` - "Check for duplicate versions" link
   2. Check "[platform]" links
      1. https://fix-specific-platform-linking.d1ywzrxfkb9wgg.amplifyapp.com/react/tools/cli/migration/transformer-migration/ - "GraphQL Transformer v2" 

#### Related GitHub issue #, if available:

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?

Which platform(s) are affected by this PR (if applicable)?

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks




      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._
